server/src/loadPhotos.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+const { connect } = require('./db');
+const flickr = require('./flickr');
+const facepp = require('./facepp');
+
+const fetchPages = async (key, fetcher, options) => {
+  const result = await fetcher(options);
+  if (result.stat === 'fail') throw result.message;
+
+  const results = [result[key]];
+  const more = result[key].page < result[key].pages;
+
+  if (!more) return results;
+
+  for (let i = 2; i <= result[key].pages; ++i) {
+    const result = await fetcher({ page: i, ...options });
+    if (result.stat === 'fail') throw result.message;
+    results.push(result[key]);
+  }
+
+  return results;
+};
+
+const fetchPhotoset = async () => {
+  const photoset = await fetchPages('photoset', flickr.fetchPhotoset);
+  return [].concat(...photoset.map(part => part.photo));
+};
+
+const fetchTagged = async () => {
+  const tagged = await fetchPages('photos', flickr.fetchPhotos, config.flickr);
+  return [].concat(...tagged.map(part => part.photo));
+};
+
+const photos = async () => {
+  console.info('Fetching images from Flickr API...');
+
+  const photoset = await fetchPhotoset();
+  const tagged = await fetchTagged();
+
+  const ids = new Set();
+  const photos = [];
+
+  photoset.forEach(img => (ids.add(img.id), photos.push(img)));
+  tagged.forEach(img => {
+    if (!ids.has(img.id)) photos.push(img);
+  });
+
+  console.info(`Got ${photos.length} images.`);
+  return photos;
+};
+
+const getEmotions = faces => {
+  const emotions = new Set();
+
+  faces.forEach(face => {
+    if (!face.attributes) return;
+    const faceEmotions = face.attributes.emotion;
+    Object.keys(faceEmotions).forEach(name => {
+      const value = faceEmotions[name];
+      if (value > config.facepp.minEmotionPercentage) emotions.add(name);
+    });
+  });
+
+  return Array.from(emotions);
+};
+
+const split = (photos, limit) => {
+  const result = [];
+  while (photos.length) {
+    result.push(photos.slice(0, limit));
+    photos = photos.slice(limit);
+  }
+  return result;
+};
+
+const detectPhoto = photo =>
+  new Promise((resolve, reject) => {
```

Comment:
I can't see if there is something that prevents you from making it async/await too.

Reply:
The specifics of Face++ API is that it can just not return request for a long time. We can't wait so much, when detecting faces on server setup, so I add `requestTimeout` option to prevent too long waiting for response. I can hardly imagine how to implement it with async/await.